Source/common/SNTFileInfo.m

```diff
@@ -112,52 +113,48 @@ - (void)hashSHA1:(NSString **)sha1 SHA256:(NSString **)sha256 {
   if (sha256) CC_SHA256_Init(&c256);
 
   int fd = self.fileHandle.fileDescriptor;
-  
+
   fcntl(fd, F_RDAHEAD, 1);
   fcntl(fd, F_RDADVISE, 1);
-
+  ssize_t bytesRead;
   for (uint64_t offset = 0; offset < _fileSize;) {
-    unsigned int readSize = 0;
-    if (offset + chunkSize > _fileSize) {
-      readSize = (unsigned int)(_fileSize - offset);
-    } else {
-      readSize = chunkSize;
-    }
-
-    unsigned int bytesRead = (unsigned int)pread(fd, &chunk, readSize, offset);
-    if (bytesRead == readSize) {
-      if (sha1) CC_SHA1_Update(&c1, chunk, readSize);
-      if (sha256) CC_SHA256_Update(&c256, chunk, readSize);
+    bytesRead = pread(fd, chunk, chunkSize, offset);
+    if (bytesRead > 0) {
+      if (sha1) CC_SHA1_Update(&c1, chunk, (CC_LONG)bytesRead);
+      if (sha256) CC_SHA256_Update(&c256, chunk, (CC_LONG)bytesRead);
       offset += bytesRead;
-    } else if (errno == EINTR) {
+    } else if (bytesRead == -1 && errno == EINTR) {
       continue;
     } else {
       return;
     }
   }
-
-  fcntl(fd, F_RDAHEAD, 0);
   fcntl(fd, F_RDADVISE, 0);
```

Comment:
Please restore the empty lines around this call, it's clearer when it's separate. Also, why remove the call to disable F_RDAHEAD?